### PR TITLE
Document the mounts.conf file

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Partial fix for https://github.com/projectatomic/buildah/issues/496

File: docs/buildah.md [hunk 1]
```diff
@@ -27,6 +27,21 @@ Print debugging information
...
+The processes running in the container during a `buildah run` or a `buildah build-using-dockerfile` can use
+this content but the content does not get committed to the final image.
+
+Usually these directories
+are used for passing secrets or credentials required by the package software to access remote package repositories.
+When you have a mounts.conf with `/usr/share/rhel/secrets:/run/secrets`, the content of /usr/share/rhel/secrets directory
+is mounted on /run/secrets inside the container.  This allows Red Hat Enterprise Linux subscriptions from the host to be used
+inside of the container.
+
+The format of the mounts.conf is /SRC:/DEST.  Users can create an `/etc/containers/mounts.conf`, to specify their
```
Comment (ripcurld) [inline docs/buildah.md]: I assume I can specify more than one mount.
I think it's worth mentioning how they are separated (by a comma, a space, a new-line)

Comment (rhatdan) [reply docs/buildah.md]: Updated

File: docs/buildah.md [hunk 2]
```diff
@@ -26,7 +26,24 @@ Print debugging information
...
+
+Usually these directories
+are used for passing secrets or credentials required by the package software to access remote package repositories.
+When you have a mounts.conf with `/usr/share/rhel/secrets:/run/secrets`, the content of /usr/share/rhel/secrets directory
+is mounted on /run/secrets inside the container.  This allows Red Hat Enterprise Linux subscriptions from the host to be used
+inside of the container.
+
+The format of the mounts.conf is the volume format /SRC:/DEST, one mount per line.
+Users can create an `/etc/containers/mounts.conf`, to specify their
+own special volumes to mount in the container.  buildah by default reads /usr/share/containers/mounts.conf and the /etc/containers/mounts.conf
```
Comment (TomSweeneyRedHat) [inline docs/buildah.md]: "buildah by default" to "Buildah by default"

File: docs/buildah.md [hunk 3]
```diff
@@ -26,7 +26,24 @@ Print debugging information
...
+if it exists. Specifying the `--default-mounts-file` overrides both `default` mounts files.
+Note this is not a volume mount, in the content of the volumes is copied into container storage not
```
Comment (TomSweeneyRedHat) [inline docs/buildah.md]: I think I'd remove 'in' from "in the content"

File: docs/buildah.md [hunk 4]
```diff
@@ -26,7 +26,24 @@ Print debugging information
 
 **--default-mounts-file**
 
-Path to default mounts file (default path: "/usr/share/containers/mounts.conf")
+Path to default mounts file (default path: "/usr/share/containers/mounts.conf") Alternate path used `/etc/containers/mounts.conf` if user created it.
```
Comment (TomSweeneyRedHat) [inline docs/buildah.md]: Suggest: ") Alternate path used ..." to ").  The alternate path is `/etc/containers/mounts.conf` if created by the user. "

File: docs/buildah.md [hunk 5]
```diff
@@ -26,7 +26,24 @@ Print debugging information
...
+
+The mounts.conf file(s) specify volume mount directories that are automatically mounted inside containers.
+The processes running in the container during a `buildah run` or a `buildah build-using-dockerfile` can use
+this content but the content does not get committed to the final image.
+
+Usually these directories
+are used for passing secrets or credentials required by the package software to access remote package repositories.
+When you have a mounts.conf with `/usr/share/rhel/secrets:/run/secrets`, the content of /usr/share/rhel/secrets directory
```
Comment (TomSweeneyRedHat) [inline docs/buildah.md]: suggest "the content of " to "the content of the"

File: docs/buildah.md [hunk 6]
```diff
@@ -26,7 +26,24 @@ Print debugging information
...
+is mounted on /run/secrets inside the container.  This allows Red Hat Enterprise Linux subscriptions from the host to be used
+inside of the container.
+
+The format of the mounts.conf is the volume format /SRC:/DEST, one mount per line.
```
Comment (TomSweeneyRedHat) [inline docs/buildah.md]: Anyway to give a more concrete example?

**Comment by rhatdan** [comment]
@TomSweeneyRedHat PTAL If this is good then merge it.

File: docs/buildah.md [hunk 7]
```diff
@@ -26,7 +26,19 @@ Print debugging information
 
 **--default-mounts-file**
 
-Path to default mounts file (default path: "/usr/share/containers/mounts.conf")
+Path to default mounts file (default path: "/usr/share/containers/mounts.conf").  Users can also create an additional conf file, `/etc/containers/mounts.conf`.
+
+The mounts.conf files specify volume mount directories that are automatically mounted inside containers when executing the `buildah run` or `buildah build-using-dockerfile` commands.  Container process can then use this content.  The volume mount content does not get committed to the final image.
+
+Usually these directories are used for passing secrets or credentials required by the package software to access remote package repositories.  For example a mounts.conf with the `/usr/share/rhel/secrets:/run/secrets like, the content of /usr/share/rhel/secrets directory is mounted on /run/secrets inside the container.  This allows Red Hat Enterprise Linux subscriptions from the host to be used
```
Comment (TomSweeneyRedHat) [inline docs/buildah.md]: "For example a mounts.conf with the `/usr/share/rhel/secrets:/run/secrets like," .  You're missing a back tick  and I think you wanted line not like?

Perhaps "For example a mounts.conf with the line `/usr/share/rhel/secrets:/run/secrets`, "

Comment (rhatdan) [reply docs/buildah.md]: Fixed.

**Comment by rhatdan** [comment]
Also fixed the table at the bottom.

**Comment by TomSweeneyRedHat** [comment]
LGTM, thanks for the touch ups and figuring out the table fun.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**